src/injections/js/bug1629926-mail.google.com-hangouts-mic-fix.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+
+/**
+ * Bug 1629926 - mail.google.com - set allow mic on hangouts iframe
+ *
+ * mail.google.com has a chain of iframes for hangouts "Make a call" interface,
+ * with a parent iframe not having allow=microphone attribute. It's resulting
+ * in calls not working since feature policy requires allow lists all the way
+ * up to the top document.
+ *
+ * Adding allow=microphone to the iframe in question makes the calls work
+ */
+
+console.info(
+  "hangouts iframe allow property was changed for compatibility reasons. See https://bugzilla.mozilla.org/show_bug.cgi?id=1503694 for details."
+);
+
+const NOTIFICATIONS_LIMIT = 150;
+
+const createObserver = callback => {
+  return new MutationObserver(callback).observe(document, {
+    childList: true,
+    subtree: true,
+  });
+};
+
+const setAllow = element => {
+  element.allow = "microphone *; autoplay *; microphone";
+};
+
+const poll = (getElementFn, onFound, limit) => {
+  const element = getElementFn();
+  if (element) {
+    onFound(element);
+    return;
+  }
+
+  let n = 0;
+  createObserver((records, observer) => {
+    const _element = getElementFn();
+    if (_element) {
+      onFound(_element);
+      observer.disconnect();
+      return;
+    }
+
+    if (n > limit) {
```

Comment:
Do we really want a limit here? Will users who leave GMail open and make many calls end up with the microphone not working?

Reply:
So this observer is waiting for the iframe element to be in the dom on initial load (once it's located, "allow" attribute is added). The limit is just to be safe, so we don't continue to poll in case the iframe with id "gth-talk-plugin-frame-id" is never found. 

I've tried making multiple calls and since the targeted iframe is never removed/modified, it works as expected

Reply:
Ah, I see. Maybe it would be better to just add a Mutation Observer right away which listens for attribute changes, and when it sees an `id` change to `gth-talk-plugin-frame-id` for an `iframe`, it adds our `allow` value? If that works, we should be able to avoid polling and just leave the MutationObserver active just in case (or disconnect it then, if the iframe seems to never be re-created by the page).